app/assets/javascripts/discourse/app/components/composer-fullscreen-prompt.js

```diff
@@ -0,0 +1,31 @@
+import GlimmerComponent from "@glimmer/component";
+import { htmlSafe } from "@ember/template";
+import I18n from "I18n";
+import { next } from "@ember/runloop";
+
+export default class ComposerFullscreenPrompt extends GlimmerComponent {
+  constructor() {
+    super(...arguments);
+    this.#setupFullscreenPrompt();
+  }
+
+  #setupFullscreenPrompt() {
+    next(() => {
+      const promptElement = document.querySelector(
+        ".composer-fullscreen-prompt"
+      );
+
+      promptElement?.addEventListener(
+        "animationend",
+        () => {
+          this.args.model.toggleProperty("showFullScreenExitPrompt");
```

Comment:
instead of passing the whole model here, can we pass just an action and call it? I don't think we'll ever need the whole composer model in this small component.